cmd/agent/subcommands/subcommands_nosecrets.go

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !secrets
+// +build !secrets
```

Comment:
Should we push down the build flag down to the `secrethelper` level ? Same for each `subcommands_*` and we could get rid of those file and have the complexity in there respective package.

WDYT ?

Reply:
(ASC-182)

Reply:
For me the important is more that when we define a new binary or import any kind of packages/dependencies the caller doesn't have to know how the internal works and what should be imported depending on the build flag.

For example: if we look at `secrets`, we could build any binaries with or without it. That's why I think we should push down the build flag.

Reply:
(I've moved this to the architecture-decisions document)